src/core/tests/ov_tensor_test.cpp

```diff
@@ -281,3 +303,94 @@ TEST_F(OVTensorTest, readRangeRoiBlob) {
         }
     }
 }
+
+struct TestParams {
+    ov::Shape src_shape;
+    ov::Strides src_strides;
+    ov::Shape dst_shape;
+    ov::Strides dst_strides;
+};
+
+struct OVTensorTestCopy : ::testing::TestWithParam<TestParams> {};
+
+TEST_P(OVTensorTestCopy, copy_to) {
+    const TestParams& p = GetParam();
+    // Source tensors
+    ov::Tensor full_src_tensor;
+    ov::Tensor src_tensor;
+    if (!p.src_strides.empty()) {
+        full_src_tensor = ov::Tensor(ov::element::i32, ov::Shape{p.src_shape[0] * p.src_strides[0]});
+        src_tensor = ov::Tensor(ov::element::i32, p.src_shape, full_src_tensor.data(), p.src_strides);
+    } else {
+        src_tensor = full_src_tensor = ov::Tensor(ov::element::i32, p.src_shape);
+    }
+    {
+        const auto origPtr = full_src_tensor.data<int32_t>();
+        ASSERT_NE(nullptr, origPtr);
+        for (size_t i = 0; i < full_src_tensor.get_size(); ++i) {
+            origPtr[i] = static_cast<int32_t>(i);
+        }
+    }
+
+    ov::Tensor full_dst_tensor;
+    ov::Tensor dst_tensor;
+    if (!p.dst_strides.empty()) {
+        full_dst_tensor = ov::Tensor(ov::element::i32, ov::Shape{p.dst_shape[0] * p.dst_strides[0]});
+        dst_tensor = ov::Tensor(ov::element::i32, p.dst_shape, full_dst_tensor.data(), p.dst_strides);
+    } else {
+        dst_tensor = full_dst_tensor = ov::Tensor(ov::element::i32, p.dst_shape);
+    }
+    {
+        const auto origPtr = full_dst_tensor.data<int32_t>();
+        ASSERT_NE(nullptr, origPtr);
+        for (size_t i = 0; i < full_dst_tensor.get_size(); ++i) {
+            origPtr[i] = static_cast<int32_t>(-1);
+        }
+    }
+
+    src_tensor.copy_to(dst_tensor);
+
+    {
+        const auto fill_data = [](const ov::Tensor& tensor) {
+            std::vector<int32_t> actual;
+            const int32_t* data = tensor.data<int32_t>();
+            auto strides = tensor.get_strides();
+            for (auto&& c : ngraph::CoordinateTransformBasic{tensor.get_shape()}) {
+                actual.emplace_back(*(data + (c[2] * strides[2] + c[1] * strides[1] + c[0] * strides[0]) /
+                                                 tensor.get_element_type().size()));
+            }
+            return actual;
+        };
+        auto source_vec = fill_data(src_tensor);
+        auto dest_vec = fill_data(dst_tensor);
+
+        ASSERT_EQ(source_vec.size(), dest_vec.size());
+
+        for (size_t i = 0; i < source_vec.size(); i++) {
+            EXPECT_EQ(source_vec[i], dest_vec[i]);
+        }
+    }
+}
+
+// clang-format off
+INSTANTIATE_TEST_SUITE_P(copy_tests,
+                         OVTensorTestCopy,
+                         ::testing::Values(
```

Comment:
should we also add an element_type as a parameter as well? E.g. to check different sizes of elements